packages/bff/src/services/clientService.ts

```diff
@@ -0,0 +1,358 @@
+        queries: {
+          offset,
+          limit,
+          userIds: userIds.join(","),
+          consumerId: requesterId,
+          name,
+          kind,
+        },
+        headers: ctx.headers,
+      });
+    },
+
+    async getClientById(
+      clientId: string,
+      ctx: WithLogger<BffAppContext>
+    ): Promise<bffApi.Client> {
+      ctx.logger.info(`Retrieve client ${clientId}`);
+
+      const client = await authorizationProcessClient.client.getClient({
+        params: { clientId },
+        headers: ctx.headers,
+      });
+      return enhanceClient(apiClients, client, ctx);
+    },
+
+    async deleteClient(
+      clientId: string,
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<void> {
+      logger.info(`Deleting client ${clientId}`);
+
+      return authorizationProcessClient.client.deleteClient(undefined, {
+        params: { clientId },
+        headers,
+      });
+    },
+
+    async removeClientPurpose(
+      clientId: string,
+      purposeId: string,
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<void> {
+      logger.info(`Removing purpose ${purposeId} from client ${clientId}`);
+
+      return authorizationProcessClient.client.removeClientPurpose(undefined, {
+        params: { clientId, purposeId },
+        headers,
+      });
+    },
+
+    async deleteClientKeyById(
+      clientId: string,
+      keyId: string,
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<void> {
+      logger.info(`Deleting key ${keyId} from client ${clientId}`);
+
+      return authorizationProcessClient.client.deleteClientKeyById(undefined, {
+        params: { clientId, keyId },
+        headers,
+      });
+    },
+
+    async removeUser(
+      clientId: string,
+      userId: string,
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<void> {
+      logger.info(`Removing user ${userId} from client ${clientId}`);
+
+      return authorizationProcessClient.client.removeUser(undefined, {
+        params: { clientId, userId },
+        headers,
+      });
+    },
+
+    async addUserToClient(
+      userId: string,
+      clientId: string,
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<void> {
+      logger.info(`Add user ${userId} to client ${clientId}`);
+
+      await authorizationProcessClient.client.addUser(undefined, {
+        params: { clientId, userId },
+        headers,
+      });
+    },
+
+    async createKeys(
+      userId: string,
+      clientId: string,
+      keySeed: bffApi.KeysSeed,
+      { logger, headers }: WithLogger<BffAppContext>
+    ): Promise<void> {
+      logger.info(`Create keys for client ${clientId}`);
+
+      const body: authorizationApi.KeysSeed = keySeed.map((seed) => ({
+        userId,
+        key: seed.key,
+        use: seed.use,
+        alg: seed.alg,
+        name: seed.name,
+        createdAt: new Date().toISOString(),
+      }));
```

Comment:
What about moving also this conversion into `apiConverter` as a dedicated function that you can then call in the `.map` here?